pkg/infrastructure/azure/network.go

```diff
@@ -0,0 +1,468 @@
+				FrontendIPConfiguration: &armnetwork.SubResource{
+					ID: to.Ptr(in.frontendIPConfigID),
+				},
+				FrontendPort: to.Ptr[int32](in.inboundNatRulePort),
+				Protocol:     to.Ptr(armnetwork.TransportProtocolTCP), // assume TCP for now
+			},
+		},
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to add inbound nat rule to load balancer: %w", err)
+	}
+	inboundNatRuleResp, err := inboundNatRulesPoller.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to add inbound nat rule to load balancer: %w", err)
+	}
+
+	return &inboundNatRuleResp.InboundNatRule, nil
+}
+
+func deleteInboundNatRule(ctx context.Context, in *inboundNatRuleInput) error {
+	inboundNatRulesClient := in.networkClientFactory.NewInboundNatRulesClient()
+	inboundNatRulesPoller, err := inboundNatRulesClient.BeginDelete(ctx,
+		in.resourceGroupName,
+		in.loadBalancerName,
+		in.inboundNatRuleName,
+		nil,
+	)
+	if err != nil {
+		return fmt.Errorf("failed to delete inbound nat rule: %w", err)
+	}
+	_, err = inboundNatRulesPoller.PollUntilDone(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("failed to delete inbound nat rule: %w", err)
+	}
+	return nil
+}
+
+func associateInboundNatRuleToInterface(ctx context.Context, in *inboundNatRuleInput) (*armnetwork.Interface, error) {
+	interfacesClient := in.networkClientFactory.NewInterfacesClient()
+	interfaceResp, err := interfacesClient.Get(ctx,
+		in.resourceGroupName,
+		in.bootstrapNicName,
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get interface: %w", err)
+	}
+	bootstrapInterface := interfaceResp.Interface
+
+	inboundNatRulesClient := in.networkClientFactory.NewInboundNatRulesClient()
+	inboundNatRulesResp, err := inboundNatRulesClient.Get(ctx,
+		in.resourceGroupName,
+		in.loadBalancerName,
+		in.inboundNatRuleName,
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get inbound nat rule: %w", err)
+	}
+	inboundNatRule := inboundNatRulesResp.InboundNatRule
+
+	for i, ipConfig := range bootstrapInterface.Properties.IPConfigurations {
+		ipConfig.Properties.LoadBalancerInboundNatRules = append(ipConfig.Properties.LoadBalancerInboundNatRules,
+			&inboundNatRule,
+		)
+		bootstrapInterface.Properties.IPConfigurations[i] = ipConfig
+	}
+
+	interfacesPollerResp, err := interfacesClient.BeginCreateOrUpdate(ctx,
+		in.resourceGroupName,
+		in.bootstrapNicName,
+		bootstrapInterface,
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to add inbound nat rule to interface: %w", err)
+	}
+
+	interfacesResp, err := interfacesPollerResp.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to add inbound nat rule to interface: %w", err)
+	}
+	return &interfacesResp.Interface, nil
+}
+
+// GetCloudConfiguration gets a cloud configuration from the cloud name and endpoint.
+func GetCloudConfiguration(cloudName aztypes.CloudEnvironment, armEndpoint string) (*cloud.Configuration, error) {
```

Comment:
For the future, this should be either grabbed from the session: https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/azure/session.go#L86-L104 + https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/azure/session.go#L133 or the code should be shared by both.

Reply:
Fixed.